src/primitives/element.js

```diff
@@ -186,7 +186,7 @@ class Element {
 
     if (!elementNames.isEmpty) {
       elements = elements.filter((element) => {
-        let parentElements = element.parents.map(element => element.element);
+        let parentElements = element.parents.map(e => e.element);
```

Comment:
I'm wondering whether `el` wouldn't be slightly better as `e` gets often used for errors or exceptions, but up to you if it's worth the change.

(Relevant to multiple places in the commit)